### ISSUE TITLE
Inline immediate `.unwrap` after `Some(_)`

### ISSUE THREAD

**Issue by KisaragiEffective** [rust-lang/rust-clippy#9816]
### What it does

Suggesting inline `Some(_)` and `.unwrap()` in continuous line.

## Before
```rs
let expr = "";
let token = Some(expr);
let token = token.unwrap();
// assert_eq!(expr, token)
```

## After
```rs
let expr = "";
let token = expr;
```

### Lint Name

unnecessary_option

### Category

complexity

### Advantage

- Remove complexibility

### Drawbacks

`None`

### Example

```rs
// let expr = "for any value and type";
let token = Some(expr);
let token = token.unwrap();
// assert_eq!(expr, token)
```

Could be written as:

```rs
let token = expr;
```

**Comment by y21** [comment]
This lint exists now: https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_literal_unwrap

**Comment by KisaragiEffective** [comment]
Yes, this should be closed because #10358 is merged and landed in today’s (>= 1.69.0) toolchain.
